File: core/workflow-pod-brain/src/main/scala/web/model/SqlServer.scala
package web.model

import com.mysql.cj.jdbc.MysqlDataSource
import config.ApplicationConf.appConfig
import org.jooq.impl.DSL
import org.jooq.{DSLContext, SQLDialect}

object SqlServer {
  private val mysqlConfig = appConfig.mysqlConfig

  val SQL_DIALECT = SQLDialect.MYSQL
  val dataSource: MysqlDataSource = new MysqlDataSource()
  dataSource.setUrl(mysqlConfig.url)
  dataSource.setUser(mysqlConfig.username)
  dataSource.setPassword(mysqlConfig.password)

  val context: DSLContext = DSL.using(dataSource, SQL_DIALECT)

  def createDSLContext(): DSLContext = {
    context
  }
}

File: core/micro-services/dao/src/main/scala/edu/uci/ics/texera/SqlServer.java
package edu.uci.ics.texera;

import com.mysql.cj.jdbc.MysqlDataSource;
import org.jooq.DSLContext;
import org.jooq.SQLDialect;
import org.jooq.impl.DSL;

public final class SqlServer {
    public static final SQLDialect SQL_DIALECT = SQLDialect.MYSQL;
    private static final MysqlDataSource dataSource;
    public static DSLContext context;

    static {
        dataSource = new MysqlDataSource();
        dataSource.setUrl(DaoConfig.jdbcUrl());
        dataSource.setUser(DaoConfig.jdbcUsername());
        dataSource.setPassword(DaoConfig.jdbcPassword());
        context = DSL.using(dataSource, SQL_DIALECT);
    }

    public static DSLContext createDSLContext() {
        return context;
    }

    public static void replaceDSLContext(DSLContext newContext){
        context = newContext;
    }
}
